aws/templates/solutions/solution_ElastiCache.ftl

```diff
@@ -0,0 +1,122 @@
+                                        "PreferredAvailabilityZones" : [
+                                            [#assign countPerZone = processorProfile.CountPerZone]
+                                            [#assign cacheZoneCount = 0]
+                                            [#list zones as zone]
+                                                [#list 1..countPerZone as i]
+                                                    [#if cacheZoneCount > 0],[/#if]
+                                                    "${zone.AWSZone}"
+                                                    [#assign cacheZoneCount += 1]
+                                                [/#list]
+                                        [/#list]
+                                        ],
+                                        "NumCacheNodes" : "${processorProfile.CountPerZone * zones?size}",
+                                    [#else]
+                                        "AZMode": "single-az",
+                                        "PreferredAvailabilityZone" : "${zones[0].AWSZone}",
+                                        "NumCacheNodes" : "${processorProfile.CountPerZone}",
+                                    [/#if]
+                                    [#if (cache.SnapshotRetentionLimit)??]
+                                        "SnapshotRetentionLimit" : ${cache.SnapshotRetentionLimit}
+                                    [/#if]
+                                    "VpcSecurityGroupIds":[
+                                        { "Ref" : "securityGroupX${tier.Id}X${component.Id}" }
+                                    ],
+                                    "Tags" : [
+                                        { "Key" : "cot:request", "Value" : "${requestReference}" },
+                                        { "Key" : "cot:configuration", "Value" : "${configurationReference}" },
+                                        { "Key" : "cot:tenant", "Value" : "${tenantId}" },
+                                        { "Key" : "cot:account", "Value" : "${accountId}" },
+                                        { "Key" : "cot:product", "Value" : "${productId}" },
+                                        { "Key" : "cot:segment", "Value" : "${segmentId}" },
+                                        { "Key" : "cot:environment", "Value" : "${environmentId}" },
+                                        { "Key" : "cot:category", "Value" : "${categoryId}" },
+                                        { "Key" : "cot:tier", "Value" : "${tier.Id}" },
+                                        { "Key" : "cot:component", "Value" : "${component.Id}" },
+                                        { "Key" : "Name", "Value" : "${productName}-${segmentName}-${tier.Name}-${component.Name}" } 
+                                    ]
+                                }
+                            }
+                            [#assign count += 1]
```

Comment:
count variable is being set but there is no check for it in Resources(definition mode) section of the template for all solutions. at the same time there is a check for it in Outputs(output mode) section. Seems to be a bug.